tree/ntuple/v7/src/RField.cxx

```diff
@@ -806,23 +806,33 @@ ROOT::Experimental::RClassField::RClassField(std::string_view fieldName, std::st
       throw RException(R__FAIL(std::string(className) + " is not supported"));
    }
 
+   if (!(fClass->ClassProperty() & kClassHasExplicitCtor))
+      fTraits |= kTraitTriviallyConstructible;
+   if (!(fClass->ClassProperty() & kClassHasExplicitDtor))
+      fTraits |= kTraitTriviallyDestructible;
+
    int i = 0;
    for (auto baseClass : ROOT::Detail::TRangeStaticCast<TBaseClass>(*fClass->GetListOfBases())) {
       TClass *c = baseClass->GetClassPointer();
       auto subField = Detail::RFieldBase::Create(std::string(kPrefixInherited) + "_" + std::to_string(i),
                                                  c->GetName()).Unwrap();
+      fTraits &= subField->GetTraits();
       Attach(std::move(subField),
 	     RSubFieldInfo{kBaseClass, static_cast<std::size_t>(baseClass->GetDelta())});
       i++;
    }
    for (auto dataMember : ROOT::Detail::TRangeStaticCast<TDataMember>(*fClass->GetListOfDataMembers())) {
-      // Skip members explicitly marked as transient by user comment
-      if (!dataMember->IsPersistent())
-         continue;
       // Skip, for instance, unscoped enum constants defined in the class
       if (dataMember->Property() & kIsStatic)
          continue;
+      // Skip members explicitly marked as transient by user comment
+      if (!dataMember->IsPersistent()) {
+         // TODO(jblomer): we could do better
+         fTraits = 0;
```

Comment:
Perhaps I would use the safer version below, as there might be other traits in the future.
```suggestion
         fTraits &= ~(kTraitTriviallyConstructible | kTraitTriviallyDestructible);
```